tester/src/test/scala/spinal/tester/scalatest/SpinalSimRandomizeTester.scala

```diff
@@ -0,0 +1,57 @@
+package spinal.tester.scalatest
+
+import org.scalatest.funsuite.AnyFunSuite
+import spinal.core._
+import spinal.core.sim._
+
+object SomeEnum extends SpinalEnum {
+  val sIdle, sStart, sData, sParity, sStop = newElement()
+}
+
+case class SimRandomizeTester() extends Component {
+  val io = new Bundle {
+    val b = in(Bool())
+    val ui = in(UInt(4 bit))
+    val uil = in(UInt(33 bit))
+    val si = in(SInt(4 bit))
+    val sil = in(UInt(33 bit))
+    val e = in(SomeEnum())
+    val uf = in(UFix(peak=8 exp, resolution = -2 exp))
+    val sf = in(SFix(peak=8 exp, resolution = -2 exp))
+  }
+  // just some content that forces a clock to be
+  // wired & available in the simulation
+  val x = RegNext(io.b)
+}
+
+class SpinalSimRandomizeTest extends AnyFunSuite {
+  val dut = SimConfig.compile { SimRandomizeTester() }
+
+  test("randomize returns the value driven next") {
+    dut.doSim("randomize returns the value driven next") { dut =>
+      dut.clockDomain.forkStimulus(10)
+
+      for (_ <- 0 to 100) {
+        val b = dut.io.b.randomize()
+        val ui = dut.io.ui.randomize()
+        val uil = dut.io.uil.randomize()
+        val si = dut.io.si.randomize()
+        val sil = dut.io.sil.randomize()
+        val e = dut.io.e.randomize()
+        val uf = dut.io.uf.randomize()
+        val sf = dut.io.sf.randomize()
+
+        dut.clockDomain.waitSampling()
```

Comment:
Could be just `sleep(10)`, to avoid tricks to add an unused clock domain?

Reply:
didn't think of that since that would be my first comb-only SpinalHDL component ;-)